Tests/GRPCHTTP2CoreTests/Server/GRPCServerStreamHandlerTests.swift

```diff
@@ -646,7 +646,15 @@ final class GRPCServerStreamHandlerTests: XCTestCase {
     XCTAssertNil(try channel.readOutbound(as: HTTP2Frame.FramePayload.self))
 
     // Receive them again. Should be a protocol violation.
-    try channel.writeInbound(HTTP2Frame.FramePayload.headers(.init(headers: clientInitialMetadata)))
+    XCTAssertThrowsError(
+      ofType: RPCError.self,
+      try channel.writeInbound(
+        HTTP2Frame.FramePayload.headers(.init(headers: clientInitialMetadata))
+      )
+    ) { error in
+      XCTAssertEqual(error.code, .unavailable)
+      XCTAssertEqual(error.message, "Stream unexpectedly closed.")
+    }
```

Comment:
@glbrntt this is what I was referring to in https://github.com/grpc/grpc-swift/pull/1953#discussion_r1658780342

We were not throwing when doing this write, we were only getting a reset stream frame.

Reply:
Gotcha. So previously the server just ignored the extra metadata? I think throwing the error here is appropriate: the server handler can't continue, the client messed up.